File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int dp[nums.size()];
        dp[0]=0;
        for (int i=1;i<nums.size();++i){
            dp[i]=INT_MAX;
            for(int j=0;j<i;++j){
                if(dp[j]!=INT_MAX and nums[j]+j>=i){
                    dp[i]=min(dp[i],dp[j]+1);
                }
            }
            // cout<<dp[i]<<" ";
        }
        if(dp[nums.size()-1]==INT_MAX) return -1;
        return dp[nums.size()-1];
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        if(len(nums)<2):
            return 0
        pos=0
        can_reach =0
        jump = 0
        while(True):
            max_reach = pos
            for x in range(pos,can_reach+1):
                max_reach = max(max_reach,x+nums[x])
            jump+=1
            if(max_reach>=len(nums)-1):
                return jump
            pos=can_reach+1
            can_reach=max_reach
        return 0